azext_iot/digitaltwins/providers/resource.py

```diff
@@ -230,6 +253,168 @@ def remove_role(self, name, assignee, role_type=None, resource_group_name=None):
+        system_identity=None,
+        user_identities=None,
+        resource_group_name=None
+    ):
+        target_instance = self.find_instance(
+            name=name, resource_group_name=resource_group_name
+        )
+
+        if not system_identity and user_identities is None:
+            raise RequiredArgumentMissingError(
+                'No identities provided to remove. Please provide system (--system) or user-assigned identities (--user).'
+            )
+
+        # Turn off system managed identity
+        if system_identity:
+            if target_instance.identity.type not in [
+                    IdentityType.system_assigned.value,
+                    IdentityType.system_assigned_user_assigned.value
+            ]:
+                raise ArgumentUsageError('Digital Twin {} is not currently using a system-assigned identity'.format(name))
+            elif target_instance.identity.type == IdentityType.system_assigned_user_assigned.value:
+                target_instance.identity.type = IdentityType.user_assigned
+            else:
+                target_instance.identity.type = IdentityType.none.value
+
+        if user_identities and target_instance.identity.user_assigned_identities:
+            # loop through user_identities to remove
+            for identity in user_identities:
+                if not target_instance.identity.user_assigned_identities.get(identity):
+                    raise ArgumentUsageError(
+                        'Digital Twin {0} is not currently using a user-assigned identity with id: {1}'.format(name, identity)
+                    )
+                del target_instance.identity.user_assigned_identities[identity]
+            if not target_instance.identity.user_assigned_identities:
+                target_instance.identity.user_assigned_identities = None
+        elif isinstance(user_identities, list):
+            target_instance.identity.user_assigned_identities = None
+
+        if target_instance.identity.type in [
+                IdentityType.system_assigned.value,
+                IdentityType.system_assigned_user_assigned.value
+        ]:
+            if getattr(target_instance.identity, 'user_assigned_identities', None):
+                target_instance.identity.type = IdentityType.system_assigned_user_assigned.value
+            else:
+                target_instance.identity.type = IdentityType.system_assigned.value
+        else:
+            if getattr(target_instance.identity, 'user_assigned_identities', None):
+                target_instance.identity.type = IdentityType.user_assigned.value
+            else:
+                target_instance.identity.type = IdentityType.none.value
+
+        try:
+            update_poller = self.mgmt_sdk.digital_twins.create_or_update(
+                resource_name=name,
+                resource_group_name=resource_group_name,
+                digital_twins_create=target_instance,
+                long_running_operation_timeout=60,
```

Comment:
Maybe we extract the timeout value in a constant somewhere appropriate and use that instead. Also do you think 60 is too short?

Reply:
This constant could also be used for the default timeout of `add_endpoint`

Reply:
I think 60 is fine - it is the default timer for the create call. We can also add a param for timeout but 60 has been getting the job done (for endpoint and so far for identity)